src/components/MessageInput/types.ts

```diff
@@ -3,6 +3,12 @@ import type { DefaultStreamChatGenerics } from '../../types/types';
 
 type AttachmentLoadingState = 'uploading' | 'finished' | 'failed';
 
+export enum AttachmentUploadState {
```

Comment:
Is this different from `UploadState`?

Reply:
I wanted to gradually replace the union type with enum that could be used by the integrators too - if they want to do some customizations. Removing the original would be breaking. WDYT, how could this be approached?

Reply:
I actually don't know, the union type doesn't sound bad, can we just keep using the union in this case?